Examples/LambdaFunctions/Sources/APIGatewayEndpoint/main.swift

```diff
@@ -0,0 +1,180 @@
+        let body3 = "If you are going to use a passage of Lorem Ipsum, you need to be sure there isn't anything embarrassing hidden in the middle of text."
+        let article1 = Article(aid: 1, title: "Article 1", body: body1, date: Date(), comments: [])
+        let article2 = Article(aid: 2, title: "Article 2", body: body2, date: Date(), comments: [])
+        let article3 = Article(aid: 4, title: "Article 3", body: body3, date: Date(), comments: [])
+        articles.append(article1)
+        articles.append(article2)
+        articles.append(article3)
+    }
+}
+
+// FIXME: Use proper Event abstractions once added to AWSLambdaRuntime
+//#if DEBUG
+//try Lambda.withLocalServer(invocationEndpoint: "/articles", {
+//    Lambda.run(APIGatewayProxyLambda())
+//} )
+//#else
+//Lambda.run(APIGatewayProxyLambda())
+//#endif
+
+Lambda.run(APIGatewayProxyLambda())
+
+// MARK: - Handler, Request and Response
+
+struct APIGatewayProxyLambda: EventLoopLambdaHandler {
+    public typealias In = APIGateway.V2.Request
+    public typealias Out = APIGateway.V2.Response
+
+    public func handle(context: Lambda.Context, event: APIGateway.V2.Request) -> EventLoopFuture<APIGateway.V2.Response> {
+        context.logger.info("[API Gateway Endpoint] Response handler")
+        // Load the test data
+        // This is where a DynamoDB table could be monitored to pull in articles instead of loading them locally.
+        let testData = TestData()
+        // Create an JSONEncoder for encoding a JSON response
+        let encoder = JSONEncoder()
+        
+        var code: UInt = 200
+        var responseBody: String = ""
+        var queryStringDict: [String: String] = [:]
+        
+        // Nested Processing function for articles
+                
+        func encodeArticles(aid: Int?, articles: [Article]) -> (UInt, String) {
+            var response = (code, "")
+            do {
+                var articleSubSet = articles
+                if let articleID = aid,
+                    articleID <= articles.count {
+                    articleSubSet = [articles[(articleID - 1)]]
+                } else if let articleID = aid,
+                    articleID > articles.count {
+                    response.0 = 200
+                    response.1 = "No articles found"
+                    return response
+                }
+                
+                context.logger.info("[API Gateway Endpoint] - articles count: \(articles.count)")
+                let jsonEncodedArticles = try encoder.encode(articleSubSet)
+                let jsonArticles = String(data: jsonEncodedArticles, encoding: .utf8)!
+                response.0 = 200
+                response.1 = jsonArticles
+                context.logger.info("[API Gateway Endpoint] - JSON articles decoded")
+            } catch {
+                response.0 = 500
+                context.logger.info("[API Gateway Endpoint] - Error encoding articles")
```

Comment:
in this case we should probably use the `error` log level. further we want probably be able to read the error.

Reply:
Good catch.  Added `error` as the log level.